### ISSUE TITLE
[Feature Request] ExecForEach: Accurately get columns by index

### ISSUE THREAD

**Issue by wanglong001** [bitfield/script#152]
Example

```
$ cat t.txt 
column1 ./go.sum
column2 ./go.mod
```

```bash
cat t.txt | awk '{print($2" "$1)}'  #  Swap 1, 2 columns
# ./go.sum column1
# ./go.mod column2
cat t.txt | awk '{system("md5sum "$2"; echo "$1)}' # Column 2 is the input for  md5sum, Column 1 is the  input for echo
# b42820e6479d5c5fa3a8e2daca66d1a1  ./go.sum
# column1
# 9659322eb77f7139a925f7745510c01e  ./go.mod
# column2
```

This is very flexible function in awk

In script, I except 
```go
script.File("t.txt").ExecForEach("md5sum $2; echo $1").Stdout()
```

**Comment by bitfield** [comment]
I'm not sure how this would work, because pipe stages communicate with one another by a stream of bytes—or lines—not discrete values. Where would `$1` and `$2` come from? 

The best thing to do here might be to write a `FilterLine` function to compute the arguments `$1` and `$2` by applying `strings.Fields` to each line, and then `ExecForEach` to invoke the shell with your suggested command.

**Comment by bitfield** [comment]
Great, so is what you're really asking for that `Column` should be able to return _multiple_ columns separated by whitespace?

**Comment by wanglong001** [comment]
> Great, so is what you're really asking for that `Column` should be able to return _multiple_ columns separated by whitespace?

Yep～ ， and the data of multiple columns can be obtained through index

**Comment by bitfield** [comment]
Right, because using the `;` character to separate two commands is a `bash` thing. `ExecForEach` uses Go's `os/exec` package, which doesn't have that functionality. Luckily, the shell does!
